### DISCUSSION TITLE
Reconnect the Linux client after restarting?

### DISCUSSION THREAD

**Discussion by TelDragon** [strongswan/strongswan#1869] [Q&A] [answered]
My client is using the Debian9 operating system.
Obtain strongswan through apt install installation.
I think it is still supported in older versions.
But I have no action on his distribution software package

```
dpkg -l | grep strongswan
ii  libstrongswan                   5.7.2-1+deb10u3                      armhf        strongSwan utility and crypto library
ii  libstrongswan-extra-plugins     5.7.2-1+deb10u3                      armhf        strongSwan utility and crypto library (extra plugins)
ii  libstrongswan-standard-plugins  5.7.2-1+deb10u3                      armhf        strongSwan utility and crypto library (standard plugins)
ii  strongswan                      5.7.2-1+deb10u3                      all          IPsec VPN solution metapackage
ii  strongswan-charon               5.7.2-1+deb10u3                      armhf        strongSwan Internet Key Exchange daemon
ii  strongswan-libcharon            5.7.2-1+deb10u3                      armhf        strongSwan charon library
ii  strongswan-pki                  5.7.2-1+deb10u3                      armhf        strongSwan IPsec client, pki command
ii  strongswan-starter              5.7.2-1+deb10u3                      armhf        strongSwan daemon starter and configuration file parser
ii  strongswan-swanctl              5.7.2-1+deb10u3                      armhf        strongSwan IPsec client, swanctl command
```

```
cat /lib/systemd/system/strongswan.service
[Unit]
Description=strongSwan IPsec IKEv1/IKEv2 daemon using ipsec.conf
After=network-online.target

[Service]
ExecStart=/usr/sbin/ipsec start --nofork
ExecReload=/usr/sbin/ipsec reload
StandardOutput=syslog
Restart=on-abnormal

[Install]
WantedBy=multi-user.target



systemctl status strongswan
● strongswan.service - strongSwan IPsec IKEv1/IKEv2 daemon using ipsec.conf
   Loaded: loaded (/lib/systemd/system/strongswan.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2023-08-28 18:39:04 CST; 13min ago
 Main PID: 9635 (starter)
    Tasks: 18 (limit: 2827)
   CGroup: /system.slice/strongswan.service
           ├─9635 /usr/lib/ipsec/starter --daemon charon --nofork
           └─9649 /usr/lib/ipsec/charon

Aug 28 18:39:23 j3r-k charon[9649]: 06[ENC] parsed INFORMATIONAL request 0 [ D ]
Aug 28 18:39:23 j3r-k charon[9649]: 06[IKE] received DELETE for IKE_SA client[1]
```

Of course, the system has already

Systemctl enable strongswan

Every time I use it, I have to manually execute it
```
Swanctl -- load all -- noprompt
Swanctl -- initiate -- child "connname"
```

I also followed,It seems that only automatic connections can be loaded. But it's not the automatic connection I want when I turn it on.

[https://docs.strongswan.org/docs/6.0/swanctl/swanctlConf.html](https://docs.strongswan.org/docs/6.0/swanctl/swanctlConf.html)

```
start_action = start
dpd_action = restart
```

Is there a better way?
I want to connect to the operating system as soon as it starts, and then reconnect when there is a network malfunction or other reason.

**Comment by tobiasbrunner** [comment, marked as answer]
> I think it is still supported in older versions.

Debian 9's EOL was in June of last year.

> Every time I use it, I have to manually execute it

Obviously, as you use the legacy systemd unit that is configured with ipsec.conf and not swanctl.conf (read the output you pasted). If you want to use swanctl, install `charon-systemd` (if available on Debian 9) and remove `strongswan-starter`.

> Is there a better way?

You probably also want to configure `keyingtries = 0`.

Alternatively, you could use trap policies and a component that creates traffic regularly to re-create the SA if needed (also recovers from fatal connection errors, which the above method does not). In this case configure `start_action=trap` and don't configure `dpd_action` or `keyingtries`.

**Reply by TelDragon** [reply]
When I searched for software package support, I received the following list of packages. I believe I have installed it completely
[https://packages.debian.org/search?suite=buster&section=all&arch=any&searchon=names&keywords=strongswan](https://packages.debian.org/search?suite=buster&section=all&arch=any&searchon=names&keywords=strongswan)

I am using the new version of Swanctl configuration
/etc/swanctl/conf. d/client.conf

Do you mean to have me delete the strongswan starter and charon system packages?I don't quite understand what you mean.

**Reply by TelDragon** [reply]
Sorry. One second after replying to your message, I noticed that you had changed your statement.
Debian does support charon system d, but it is not a mandatory dependency installation
[https://packages.debian.org/buster/charon-systemd](https://packages.debian.org/buster/charon-systemd)

```
apt install charon-systemd
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  charon-systemd
0 upgraded, 1 newly installed, 0 to remove and 205 not upgraded.
Need to get 93.0 kB of archives.
After this operation, 203 kB of additional disk space will be used.
Get:1 http://mirrors.ustc.edu.cn/raspbian/raspbian buster/main armhf charon-systemd armhf 5.7.2-1+deb10u3 [93.0 kB]
Fetched 93.0 kB in 3s (28.4 kB/s)   
Selecting previously unselected package charon-systemd.
(Reading database ... 57421 files and directories currently installed.)
Preparing to unpack .../charon-systemd_5.7.2-1+deb10u3_armhf.deb ...
Unpacking charon-systemd (5.7.2-1+deb10u3) ...
Setting up charon-systemd (5.7.2-1+deb10u3) ...
Created symlink /etc/systemd/system/multi-user.target.wants/strongswan-swanctl.service → /lib/systemd/system/strongswan-swanctl.service.

cat /lib/systemd/system/strongswan-swanctl.service
[Unit]
Description=strongSwan IPsec IKEv1/IKEv2 daemon using swanctl
After=network-online.target

[Service]
Type=notify
ExecStart=/usr/sbin/charon-systemd
ExecStartPost=/usr/sbin/swanctl --load-all --noprompt
ExecReload=/usr/sbin/swanctl --reload
ExecReload=/usr/sbin/swanctl --load-all --noprompt
Restart=on-abnormal

[Install]
WantedBy=multi-user.target


apt-get remove strongswan-starter
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages will be REMOVED:
  strongswan strongswan-charon strongswan-starter
0 upgraded, 0 newly installed, 3 to remove and 205 not upgraded.
After this operation, 979 kB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 57432 files and directories currently installed.)
Removing strongswan (5.7.2-1+deb10u3) ...
Removing strongswan-charon (5.7.2-1+deb10u3) ...
Removing strongswan-starter (5.7.2-1+deb10u3) ...
Processing triggers for man-db (2.8.5-2) ...
```

**Comment by TelDragon** [comment]
That's great!!!
Now it's automatic
Systemctl enable strongswan swantl


As soon as the device is started, it will automatically connect.
Or manually
Systemctl start strongswan swantl
It will connect!

**Reply by TelDragon** [reply]
Oh, now it only works on manually executing commands such as system ctl restart strong swan swantl or system ctl start strong swan swantl.

But for example, network interruption, or client restart, server interruption. It doesn't work. What else do I need to do?
